### DISCUSSION TITLE
How to use gridTemplateAreas ?

### DISCUSSION THREAD

**Discussion by ardaberrun** [chakra-ui/chakra-ui#3098] [Q&A] [answered]
I used GridItem, it didn't work. I gave them height and width, it didn't work. I removed the gridArea property. I added a class myself, but not. I converted the GridItems to Box and did what I did before, again it didn't work. Likewise, I tried templateAreas in every variation I could think of but failed. Where am I making a mistake? How can use it ? 

this is my code:

```
    <Grid
      h="100vh"
      p="20px"
      templateColumns={{ base: "repeat(6,1fr)" }}
      gap={4}
      templateAreas= {{base:["header header header header header header" ,"card map map map card card" ,"scrollcard map map map scrollcard scrollcard" ,"scrollcard map map map scrollcard scrollcard", "footer footer footer footer footer footer"]}}
  // templateAreas= {{base:`"header header header header header header" "card map map map card card" "scrollcard map map map //scrollcard scrollcard" "scrollcard map map map scrollcard scrollcard" "footer footer footer footer footer footer"`}}
      >
        <Box  w="100%" h="100px" gridArea="header" className="grid-item" bg="yellow.100"/>    
        <Box   bg="yellow.100"  gridArea={{base:"card"}} className="grid-item2"/>    
        <Box   bg="yellow.400" gridArea={{base:`scrollcard`}} className="grid-item3"/>    
        <GridItem  w="100%" h="100px" gridArea="map" className="grid-item" bg="yellow.100"/>   

        <GridItem  gridArea="footer" className="grid-item" bg="yellow.100"/>
     
    </Grid>
```

**Comment by chasinhues** [comment, marked as answer]
Yeah using `templateAreas` as a responsive prop is a tricky one, good fun though haha! 

First thing I can see is that it looks like you're splitting areas with the same name. Try being more specific with the names of your areas. Instead of `card map map map card card`, prefer something like `leftCard map map map rightCard rightCard`. 

Judging by your code, my understanding of the layout you're trying to achieve is something like this:
| 1fr |1fr |1fr |1fr |1fr |1fr |
| --- | --- | --- | --- | --- | --- |
| Header |Header |Header |Header |Header |Header |
| leftCard | map | map | map | rightCard | rightCard |
| leftScrollCard | map | map | map | rightScrollCard | rightScrollCard |
| leftScrollCard | map | map | map | rightScrollCard | rightScrollCard |
| Footer| Footer| Footer| Footer| Footer| Footer|

Just refining this down a bit, this layout can be constructed with only 3 columns and 4 rows if you adjust the size of those columns and rows on the grid. 
| 1fr |3fr |2fr |
| --- | --- | --- | 
| Header |Header |Header |
| leftCard | map| rightCard | 
| leftScrollCard | map | rightScrollCard | 
| Footer| Footer| Footer|

For the `gridTemplateAreas` prop, use backticks to contain the grid template areas string you provide. In your grid template areas string, each row of areas you define is contained within its own string. Example:
```jsx
gridTemplateAreas=`'first line' 'second line'`
```

### Code Demo
This demo gives you an example of using responsive styles for the `gridTemplateAreas` prop.
https://codesandbox.io/s/grid-template-areas-demo-lelf4?file=/src/App.tsx

**Reply by ardaberrun** [reply]
Thank you for helping. Im thankfyl to you..  :)
